### PR TITLE
[example]kafka example

### PR DESCRIPTION
**Pull request by qyyy**
Change the output logic code from Consumer.py to start.sh.
Remove the hard code in Consumer.sh.
Modify some configuration file.

### CODE REVIEW

**Comment by hwuu** [comment]
Please give this PR a name that is easier to understand.

File: examples/kafka/start.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#startup zookeeper
+cd /root/zookeeper-3.4.12/conf
+sed -i '14d' zoo.cfg
+sed -i "13a clientPort=$PAI_CONTAINER_HOST_zookeeper_PORT_LIST" zoo.cfg
+cd /root/zookeeper-3.4.12/bin
+./zkServer.sh start
+sleep 5s
```
Comment (hwuu) [inline examples/kafka/start.sh]: Why sleep for 5s?

File: examples/kafka/start.sh [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+sed -i "31a listeners=PLAINTEXT://localhost:$PAI_CONTAINER_HOST_kafka_PORT_LIST" server.properties
+cd /root/kafka_2.11-1.1.0
+service="zookeeper"
+while ! ps -ef | grep $service | egrep -v grep >/dev/null
+do
+    sleep 5s
+done
+./bin/kafka-server-start.sh config/server.properties 1>/dev/null 2>&1 &
+pid=$!
+sleep 30s
```
Comment (hwuu) [inline examples/kafka/start.sh]: Why sleep for 30s?

Comment (qyyy) [reply examples/kafka/start.sh]: To make sure that kafka server has been started.

File: examples/kafka/start.sh (hunk 1 above)
Comment (qyyy) [reply examples/kafka/start.sh]: To make sure that zookeeper server has been started.

File: examples/kafka/start.sh [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+done
+
+#create topic
+#./bin/kafka-topics.sh --create --zookeeper localhost:2181 --replication-factor 1 --partitions 1 --topic test
+#sleep 5s
+
+#run python example
+cd /root/python-kafka-test
+sed -i "3a host='localhost:$PAI_CONTAINER_HOST_kafka_PORT_LIST'" Producer.py
+sed -i "3a host='localhost:$PAI_CONTAINER_HOST_kafka_PORT_LIST'" Consumer.py
```
Comment (Asakuri) [inline examples/kafka/start.sh]: it's better to make host as a script parameter.

File: examples/kafka/start.sh [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+cd /root/zookeeper-3.4.12/conf
+sed -i '14d' zoo.cfg
+sed -i "13a clientPort=$PAI_CONTAINER_HOST_zookeeper_PORT_LIST" zoo.cfg
+cd /root/zookeeper-3.4.12/bin
+./zkServer.sh start
+sleep 5s
+
+#startup kafka
+cd /root/kafka_2.11-1.1.0/config
+sed -i '123d' server.properties
```
Comment (Asakuri) [inline examples/kafka/start.sh]: you'd better not use line number hard code. why not use regex to substitue strings?

File: examples/kafka/README.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+If you could use kafka correctly in your own environment, of course computer or server, you could:
+
+1. build the environment into a docker image.
+
+2. push it into a public registry.
+
+3. submit your job on PAI webportal.
+
+Note:
+Now(2018-6-26), we can't use static port and dynamic port at the same time, and the job would add two dynamic ports(ssh and http) automatically. So, we could only use dynamic port to support zookeeper and kafka instead of static port. Then the way to use dynamic port is following:
```
Comment (abuccts) [inline examples/kafka/README.md]: I don't think this note is necessary, just change to the following?
> To use ports allocated by PAI:

Comment (qyyy) [reply examples/kafka/README.md]: If the user just read the official document of zookeeper and kafka, he might be possible to use the static ports. So, I think we should tell him not to use them and why.

File: examples/kafka/README.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+1. Delete the port setting sentences in "zoo.cfg" and "server.properties".
+
+2. Add the sentences you have deleted by using command "sed" and environment variable:
+
+`sed -i "13a clientPort=$PAI_CONTAINER_HOST_zookeeper_PORT_LIST" zoo.cfg`
+
+`sed -i "123a zookeeper.connect=localhost:$PAI_CONTAINER_HOST_zookeeper_PORT_LIST" server.properties`
+
+`sed -i "31a listeners=PLAINTEXT://localhost:$PAI_CONTAINER_HOST_kafka_PORT_LIST" server.properties`
```
Comment (abuccts) [inline examples/kafka/README.md]: Please refer to [an example](https://github.com/Microsoft/pai/blob/master/examples/cntk/README.md#basic-environment) for the format. Use three backticks for a code block and use four spaces for indent.

**Pull request merged**